doc/monitoring_containers/monitoring_containers.md

```diff
@@ -0,0 +1,276 @@
+# Monitoring and Auto-Mitigating the Unhealthy of Docker Containers in SONiC
+
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+* [List of Tables](#list-of-tables)
+* [Revision](#revision)
+* [About this Manual](#about-this-manual)
+* [Scope](#scope)
+* [Defintions/Abbreviation](#definitionsabbreviation)
+* [1 Feature Overview](#1-feature-overview)
+    - [1.1 Requirements](#11-requirements)
+        - [1.1.1 Functional Requirements](#111-functional-requirements)
+        - [1.1.2 Configuration and Management Requirements](#112-configuration-and-management-requirements)
+        - [1.1.3 Scalability Requirements](#113-scalability-requirements)
+    - [1.2 Design](#12-design)
+        - [1.2.1 Basic Approach](#121-basic-approach)
+* [2 Functionality](#2-functionality)
+    - [2.1 Target Deployment Use Cases](#21-target-deployment-use-cases)
+    - [2.2 Functional Description](#22-functional-description)
+        - [2.2.1 Monitoring Critical Processes](#221-monitoring-critical-processes)
+        - [2.2.2 Monitoring Critical Resource Usage](#222-monitoring-critical-resource-usage)
+        - [2.2.3 Auto-restart Docker Container](#223-auto-restart-docker-container)
+            - [2.2.3.1 CLI (and usage example)](#2231-cli-and-usage-example)
+                - [2.2.3.1.1 Show the Status of Auto-restart](#22311-show-the-status-of-auto-restart)
+                - [2.2.3.1.2 Configure the Status of Auto-restart](#22312-configure-the-status-of-auto-restart)
+                - [2.2.3.1.3 CONTAINER_FEATURE Table](#22313-container_feature-table)
+
+# List of Tables
+* [Table 1: Abbreviations](#definitionsabbreviation)
+
+# Revision
+| Rev |    Date    |          Author        |     Change Description    |
+|:---:|:----------:|:----------------------:|---------------------------|
+| 0.1 | 02/18/2020 | Yong Zhao, Joe Leveque |      Initial version      |
+
+# About this Manual
+This document presents the design and implementation of feature to monitor and auto-mitigate
+the unhealthy of docker containers in SONiC.
```

Comment:
This section seems redundant with "Scope", and this isn't really a "manual", so remove this section.

Reply:
Removed.